src/pybind/rbd/rbd.pyx

```
@@ -1360,9 +1364,11 @@ cdef class Image(object):
     cdef object ioctx
     cdef rados_ioctx_t _ioctx
 
-    def __init__(self, ioctx, name, snapshot=None, read_only=False):
+    def __init__(self, ioctx, name=None, snapshot=None, read_only=False, img_id=None):
         """
         Open the image at the given snapshot.
+        If image name and image id are both specified, utilize image name.
```

Nit: I would just throw an exception since it shouldn't be possible